core/src/main/java/com/linecorp/armeria/common/ResponseCompleteException.java

```diff
@@ -26,16 +26,23 @@ public final class ResponseCompleteException extends CancellationException {
 
     private static final long serialVersionUID = 6090278381004263949L;
 
-    private static final ResponseCompleteException INSTANCE = new ResponseCompleteException();
+    private static final ResponseCompleteException INSTANCE = new ResponseCompleteException(false);
 
     /**
      * Returns the singleton {@link ResponseCompleteException}.
      */
     public static ResponseCompleteException get() {
-        return INSTANCE;
+        if (Flags.verboseResponses() &&
```

Comment:
I'm unsure if `Flags.verboseResponses()` is a good flag to fill the stack traces. Suggest to me if there are other ideas.

Reply:
Question) Why do we check `verboseResponses` in addition to `verboseExceptionSampler`? Isn't it enough to just fill in the stack trace if `verboseExceptionSampler` is sampled?

Reply:
The default sampler fills the stack trace 10 times per second for each exception. In contrast, `ResponseCompleteException` does not expose the stack trace by default. I tried using the verboseResponses flags to differentiate between the two default behaviors, but it seems like a bad choice.

It would be better to 1) change the sampler spec of the default `Sampler` to not sample `ResponseCompleteException` or to 2) apply the same spec to `ResponseCompleteException`. I slightly prefer 1) option.

Reply:
> I tried using the verboseResponses flags to differentiate between the two default behaviors, but it seems like a bad choice.

I understood that you would like to differentiate `ResponseCompleteException` sampling so that the stacktrace is always filled (as opposed to 10 times per second).

May I ask why though? Is it difficult to debug failures due to `ResponseCompleteException`? I thought it was just thrown when the response is completed, but the request isn't completed yet.

Reply:
> I understood that you would like to differentiate ResponseCompleteException sampling so that the stacktrace is always filled (as opposed to 10 times per second).

My intention was
- Original: ResponseCompleteException is never sampled.
- Proposal: Keep the original behavior when verboseResponse=false but ExceptionSampler could be respected when verboseResponse=true.

> I'm not sure I understand why ResponseCompleteException needs special treatment.

I thought performance was the main reason. As a response was sent and completed, ResponseCompleteException could be regarded as a safe exception.

Reply:
I see.

> I'm not sure I understand why ResponseCompleteException needs special treatment.
> I thought performance was the main reason. As a response was sent and completed, ResponseCompleteException could be regarded as a safe exception.

I thought compared with `ClosedSessionException`, `WriteTimeoutException`, `ResponseTimeoutException`, etc... `ResponseCompleteException` doesn't occur as often. 

Or are you planning on also applying the `verboseResponse` condition on the rest of these exceptions as well? I guess my main concern is that from a user's standpoint, it is unclear how to enable stacktraces unless the actual implementation is checked. If the behavior for each exception type is different, I think even I wouldn't be sure until I check the code.

If `ResponseCompleteException` merits different treatment from the other exception types, I think it would help if the reasoning is documented at least in this PR. Let me assume this will be done, and go ahead and approve this PR for now.

Reply:
I think we need an easier solution to control sampling rate for each exception. Let me resume this work with a new idea.

Reply:
> May I ask why though? Is it difficult to debug failures due to ResponseCompleteException? I'm not sure I understand why ResponseCompleteException needs special treatment.
I thought compared with ClosedSessionException, WriteTimeoutException, ResponseTimeoutException, etc... ResponseCompleteException doesn't occur as often.

Yeah, actually that was the reason that I've implemented `ResponseCompleteException` in this way.
Because it's raised when the response is complete while the corresponding request isn't completed, there's not so much information we can get from the stack trace. 
Especially when `requestAutoAbortDelayMillis` is greater than 0:
https://github.com/line/armeria/blob/7c21957f42c1791e48c4980c814ac54047444a67/core/src/main/java/com/linecorp/armeria/client/HttpResponseWrapper.java#L233-L234
Having said this, I think there's no harm in adding the stack trace because this exception rarely occurs.

Reply:
Removed `verboseResponse() = true` condition. The exception sampler is now only used.